src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -276,8 +276,13 @@ export class ProgrammingExerciseUpdateComponent implements AfterViewInit, OnDest
             this.programmingExercise.maxStaticCodeAnalysisPenalty = undefined;
         }
 
-        // Automatically enable the checkout of the solution repository for Haskell exercises
-        this.programmingExercise.checkoutSolutionRepository = this.checkoutSolutionRepositoryAllowed && language === ProgrammingLanguage.HASKELL;
+        if (language == ProgrammingLanguage.HASKELL) {
+            // Automatically enable the checkout of the solution repository for Haskell exercises
+            this.programmingExercise.checkoutSolutionRepository = this.checkoutSolutionRepositoryAllowed;
+        }
```

Comment:
Question to the OCAML-Instructors (@just-max and @jerhard):
Do you always need the solution repository for OCAML exercises? Would it make sense to also activate this option per default for OCAML?

Reply:
Yes, the OCaml exercise template and all the exercises we use always need the solution checkout. Activating by default would make sense :+1: